src/main/java/edu/cornell/kfs/coa/batch/service/impl/CopyLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+                    legacyAccountAttachment.getId(), legacyAccountAttachment.getKfsChartCode(),
+                    legacyAccountAttachment.getKfsAccountNumber());
+        } catch (final Exception e) {
+            LOG.error("copyLegacyAccountAttachmentToKfs, Failed to copy attachment: {}", legacyAccountAttachment, e);
+            throw (e instanceof RuntimeException) ? (RuntimeException) e : new RuntimeException(e);
+        }
+    }
+
+    private Account getKfsAccountForAttachment(final LegacyAccountAttachment legacyAccountAttachment) {
+        final Account account = accountService.getByPrimaryId(legacyAccountAttachment.getKfsChartCode(),
+                legacyAccountAttachment.getKfsAccountNumber());
+        if (ObjectUtils.isNull(account)) {
+            throw new RuntimeException("Account '" + legacyAccountAttachment.getKfsChartCode()
+                    + "-" + legacyAccountAttachment.getKfsAccountNumber() + "' does not exist");
+        }
+        return account;
+    }
+
+    private AccountingXmlDocumentBackupLink createBackupLinkForAttachmentDownload(
+            final LegacyAccountAttachment legacyAccountAttachment) throws URISyntaxException {
+        final AccountingXmlDocumentBackupLink backupLink = new AccountingXmlDocumentBackupLink();
+        backupLink.setCredentialGroupCode(CuCoaBatchConstants.DFA_ATTACHMENTS_GROUP_CODE);
+        backupLink.setFileName(legacyAccountAttachment.getFileName());
+        backupLink.setDescription(createNoteText(legacyAccountAttachment));
+        backupLink.setLinkUrl(buildAttachmentDownloadUrl(legacyAccountAttachment));
+        return backupLink;
+    }
+
+    private void saveAttachmentToKfs(final Attachment attachment, final Account account,
+            final String noteText) throws IOException {
+        final Person systemUser = personService.getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        final Note note = new Note();
+        note.setAuthorUniversalIdentifier(systemUser.getPrincipalId());
+        note.setNoteTypeCode(NoteType.BUSINESS_OBJECT.getCode());
+        note.setRemoteObjectIdentifier(account.getObjectId());
+        note.setNoteText(noteText);
+        note.setNotePostedTimestampToCurrent();
+        note.setAttachment(attachment);
+        noteService.save(note);
+    }
+
+    private String createNoteText(final LegacyAccountAttachment legacyAccountAttachment) {
+        final int noteTextMaxLength = dataDictionaryService.getAttributeMaxLength(Note.class,
+                KRADConstants.NOTE_TEXT_PROPERTY_NAME);
+        final String noteTemplate = configurationService.getPropertyValueAsString(
+                CuCOAKeyConstants.LEGACY_ACCOUNT_ATTACHMENT_NOTE_TEXT_TEMPLATE);
+        final String noteText = MessageFormat.format(noteTemplate, legacyAccountAttachment.getLegacyAccountCode(),
+                legacyAccountAttachment.getAddedBy(), legacyAccountAttachment.getFileDescription());
+        return StringUtils.left(noteText, noteTextMaxLength);
```

Comment:
I think it is very unlikely the note text will be too long.  KFS Notes support 800 characters and the longes file description inn temp_acct_attach_for_copying is 75 characters.
However, if the length is too long, I think it would be best to error and let the functionals edit the file description.

Reply:
The file description is coming from a generated table that is not displayed in the KFS UI; there's no place for the functionals to "edit" the description unless they run SQL (or we run SQL for them).

I could change this to error out in too-long situations, but then we'd need manual SQL intervention to fix the file description. Thoughts?

Reply:
Yes, I agree with that